### PR TITLE
feature/VDX-279: added documents

### PR DESCRIPTION
**Pull request by sksadjad**
added the following documents:
- overview.md: an overview of the projects
- agent-setup.md: informations about the agent module and how to set it up
- vci-front-end.md: informations about the vci-frontend module and how to set it up 
- credential-branding.md: information about credential branding
- vci-flow.puml: a high-level diagram describing our vci procedure

### CODE REVIEW

File: documents/agent-setup.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+## Agent setup
```
Comment (BtencateSphereon) [inline documents/agent-setup.md]: would be nice to add the same structure we do for most projects. I would just look at the readme's in the ssi-sdk.

so add a header here above ## Agent setup


<h1 align="center">
  <br>
  <a href="https://www.sphereon.com">Sphereon</a>
  <br>Agent
  <br>
</h1>

ADD DESCRIPTION FOR THE PROJECT

File: documents/agent-setup.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+In this document we're going to show you how to setup your agent module.
+Agent module uses the config values that are specified in your `.env` file. In Your `.env` file, you have a property `CONF_PATH` based on this value, agent will setup itself. So let's go through them together. If you open your `packages/agent/conf` folder and head to `examples`, you'll see 6 folders:
```
Comment (BtencateSphereon) [inline documents/agent-setup.md]: Your -> your

Comment (BtencateSphereon) [inline documents/agent-setup.md]: we could add a table with all possible env options and descriptions what they are used for. Instead of explaining options in a piece of text

File: documents/agent-setup.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+- dids
+- oid4vci_metadata
+- oid4vci_options
+- oid4vp_options
+- presentation_definitions
+- templates
+
+#### dids
+This folder contains did documents that will be used throughout the vci process.
```
Comment (BtencateSphereon) [inline documents/agent-setup.md]: we have an issuer and verifier

File: documents/agent-setup.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+#### oid4vci_metadata
+The json files in this folder contain VCI metadata from [VCI spec](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html), and some other agent specific data:
+```ts
+{
+  correlationId: string
+  overwriteExisting: boolean
+  metadata: VCIMetadata
+}
+```
+`correlationId` and `overwriteExisting` are used for setting up agent's store. For a more detailed account of VCI inner works in Sphereon libraries, you can visit our underlying library ([SSI-SDK](https://github.com/Sphereon-Opensource/SSI-SDK.git))
```
Comment (BtencateSphereon) [inline documents/agent-setup.md]: this does not really explain what they are for specifically. if i read this i still do not know the difference between correlationId and overwriteExisting. or what they are actually used for.

also think about formatting this information. paragraphs are not always the best option to display such information. a table might be better

File: documents/agent-setup.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+#### oid4vp_options
+You have to create an oid4vp option of your own. The important bit here is to have a file for this id in your `presentation_definitions`.
```
Comment (BtencateSphereon) [inline documents/agent-setup.md]: does not explain what oid4vp_options is for and what it does. etc

File: documents/agent-setup.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+#### presentation_definitions
+Here you have to have your own `presentation_definition`. For a detailed account of Presentation Definitions and how they work, you can read [DIF Presentation Exchange](https://identity.foundation/presentation-exchange/) this library is using [Sphereon's PEX](https://github.com/Sphereon-Opensource/PEX) as the underlying library for presentation exchange.
```
Comment (BtencateSphereon) [inline documents/agent-setup.md]: does not explain what presentation_definitions is for, how to set them up in the configuration, etc

File: documents/agent-setup.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+### templates
+We're using `handlebars` for populating our credentials. You can take a look at our `.hbs` files and see how we're populating these templates as an example:
```
Comment (BtencateSphereon) [inline documents/agent-setup.md]: add link for handlebars documentation

File: documents/agent-setup.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  overwriteExisting: boolean
+  metadata: VCIMetadata
+}
+```
+`correlationId` and `overwriteExisting` are used for setting up agent's store. For a more detailed account of VCI inner works in Sphereon libraries, you can visit our underlying library ([SSI-SDK](https://github.com/Sphereon-Opensource/SSI-SDK.git))
+#### oid4vp_options
+You have to create an oid4vp option of your own. The important bit here is to have a file for this id in your `presentation_definitions`.
+#### presentation_definitions
+Here you have to have your own `presentation_definition`. For a detailed account of Presentation Definitions and how they work, you can read [DIF Presentation Exchange](https://identity.foundation/presentation-exchange/) this library is using [Sphereon's PEX](https://github.com/Sphereon-Opensource/PEX) as the underlying library for presentation exchange.
+### templates
```
Comment (BtencateSphereon) [inline documents/agent-setup.md]: does not explain how to set them up within the configuration

File: documents/agent-setup.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        "Voornaam": "{{Voornaam}}",
+        "Achternaam": "{{Achternaam}}",
+        "E-mail": "{{email}}",
+        "type": "Sphereon Guest"
+    }
+}
+```
+
+## Setting up your own agent
+To setup your own agent, you need to create all the aforementioned folder in your desired path (`CONF_PATH` in your `.env`). Then you need to change the values according to your own desired behaviour (credential metadata, presentation definition, etc). Once you've done all these steps, you can run your agent with `start:dev` or `start:prod` commands.
```
Comment (BtencateSphereon) [inline documents/agent-setup.md]: just do config

that is all i am reading, but it is not explaining how

Comment (BtencateSphereon) [inline documents/agent-setup.md]: folder -> folders

**Review by BtencateSphereon — CHANGES_REQUESTED** [review]
the readme is so superficial, it is not really explaining anything tbh

**Comment by nklomp** [comment]
Agreed, way too shallow to be usable by someone. Then it all of a sudden goes into way too much details about code handling the branding. Someone setting up a demo certainly should not be distracted by how our wallets or SDKs internally work to render/manage the branding.  The Readme should be about what you need to do to actually issue a branded credential, like providing a rectangular image without rounded corners, a logo, some locale texts etc.

On 31 Oct 2023 14:00, Bram ten Cate ***@***.***> wrote:

@BtencateSphereon requested changes on this pull request.

the readme is so superficial, it is not really explaining anything tbh

File: documents/agent-setup.md (hunk 1 above)
Comment (sksadjad) [reply documents/agent-setup.md]: added the header for all the markdowns

File: documents/agent-setup.md (hunk 2 above)
Comment (sksadjad) [reply documents/agent-setup.md]: changed

Comment (sksadjad) [reply documents/agent-setup.md]: added a table here and everywhere that we have a lot of variables

File: documents/agent-setup.md (hunk 3 above)
Comment (sksadjad) [reply documents/agent-setup.md]: changed the text describing this

File: documents/agent-setup.md (hunk 4 above)
Comment (sksadjad) [reply documents/agent-setup.md]: modified this file. we have a table and a more in-depth description

File: documents/agent-setup.md (hunk 5 above)
Comment (sksadjad) [reply documents/agent-setup.md]: modified this file. we have a table and a more in-depth description

File: documents/agent-setup.md (hunk 6 above)
Comment (sksadjad) [reply documents/agent-setup.md]: modified this file. we have a table and a more in-depth description

File: documents/agent-setup.md (hunk 7 above)
Comment (sksadjad) [reply documents/agent-setup.md]: added link to handlebars

File: documents/agent-setup.md (hunk 8 above)
Comment (sksadjad) [reply documents/agent-setup.md]: modified this file.added a more in-depth description for this part

File: documents/agent-setup.md (hunk 9 above)
Comment (sksadjad) [reply documents/agent-setup.md]: modified the file and added more in-depth description.

Comment (sksadjad) [reply documents/agent-setup.md]: changed

**Comment by sksadjad** [comment]
@BtencateSphereon and @nklomp modified the docs.

File: documents/agent-setup.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+2. **oid4vci_metadata**: In this folder, you'll find JSON files that detail the metadata required by the [VCI specification](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html) and some values that are essential for configuring this agent. Following is a table of values that should be set for a json file inside `oid4vci_metadata`:
+
+| Variable              | Sub-Variable                 | Description                                                                                                                                                                                                     |
+|-----------------------|------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `correlationId`       |                              | A unique identifier used to track the transaction within the system.                                                                                                                                            |
```
Comment (nklomp) [inline documents/agent-setup.md]: Huh, this is not about transactions at all. This is about mapping different configuration files onto the same internal issuer or verifier. So a correlationId between an options file and for instance a metadata file endup in the same issuer/verifier instance and DB

File: documents/agent-setup.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+| `overwriteExisting`   |                              | Determines whether to replace existing data with new data.                                                                                                                                                      |
```
Comment (nklomp) [inline documents/agent-setup.md]: Be more specific than data. We are taking about oid4vci metadata. So this will replace the existing oid4vci metadata stored in the database with new values coming from the metadata configuration file

File: documents/agent-setup.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+| `metadata`            |                              | Contains configuration details and specifications for credential issuance.                                                                                                                                      |
+|                       | `credential_issuer`          | The entity that issues and signs the credential.                                                                                                                                                                |
```
Comment (nklomp) [inline documents/agent-setup.md]: entity is way to broad. Please lookup the actual definition in the spec. This is a URL

File: documents/agent-setup.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+|                       | `credential_endpoint`        | The URL where the credential can be accessed.                                                                                                                                                                   |
```
Comment (nklomp) [inline documents/agent-setup.md]: Please lookup in spec

File: documents/agent-setup.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+|                    | `responseMode`        | (optional) Specifies the method that the RP uses to receive the response from the OpenID Provider.      |
+|                    | `supportedVersions`   | (optional) Lists the versions supported by the RP.                                                      |
+|                    | `sessionManager`      | (optional) An instance of IRPSessionManager from Sphereon's SIOP-OID4VP library managing session state. |
+|                    | `expiresIn`           | (optional) Defines the duration after which the RP's request or session should expire.                  |
+|                    | `eventEmitter`        | (optional) An object used to handle events, enabling asynchronous event-driven programming.             |
+|                    | `didOpts`             | Options related to the Decentralized Identifier (DID) of the verifier.                                  |
+
+5. **presentation_definitions**: The content of this directory is essential for defining how the agent understands and processes presentation requests.
```
Comment (nklomp) [inline documents/agent-setup.md]: Please mention OID4VP in this. Be more precise in how you describe things

File: documents/agent-setup.md [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+
+6. **templates**: This directory includes templates for the dynamic generation of content, We're populating the credential using [handlebars](https://handlebarsjs.com/). You can find examples of this usage in this directory. Here is an example of such usage:
```
Comment (nklomp) [inline documents/agent-setup.md]: not content, we are talking about how to generate Verifiable Credentials.

File: documents/agent-setup.md [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+           "type": "Sphereon Guest"
+       }
+   }
+   ```
+
+Each directory plays a critical role in the configuration of the agent and the successful deployment of the verifiable credential issuance ecosystem within your application. It is essential to carefully craft the contents of these directories in alignment with your specific use cases and the standards of the decentralized identity space.
+
+### .env variables
+
+Your `.env` file should contain the following variables with values tailored to your setup:
```
Comment (nklomp) [inline documents/agent-setup.md]: .env.local

File: documents/credential-branding.md [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+        - **Alt Text (OPTIONAL):** A string value serving as alternative text for the logo image.
+
+    - **Description (OPTIONAL):** A string value providing a description of the credential.
+
+    - **Background Color (OPTIONAL):** A string value representing the background color of the credential, specified using numerical color values defined in CSS Color Module Level 3.
+
+    - **Text Color (OPTIONAL):** A string value representing the text color of the credential, specified using numerical color values defined in CSS Color Module Level 3.
+
+## Branding in real-world
+We're using branding in this project. So in order to get familiar with it, you can visit one of our config json files in `packages/agent/conf/examples/oid4vci_metadata`. You can find the branding related information (as the spec suggests in `credentials_supported` field of the metadata)
```
Comment (nklomp) [inline documents/credential-branding.md]: Make links out of these instead of highlights

File: documents/credential-branding.md [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+    }
+  ],
+  ...
+}
+```
+## Implementation
+
+Credential branding is typically specified by credential issuers in their metadata. They define how each credential they issue should be presented, including its name, logo, and visual style. Credential holders and verifiers use this branding information to ensure a consistent and recognizable display of the credential.
+
+In code, credential branding can be accessed and displayed using appropriate methods. For example, following code (in our oid4vci-demo-frontend) shows how to read and display credential branding. The `credentialBranding` object is populated with information extracted from the `credentials_supported` metadata parameter, which includes display properties and other branding-related details.
```
Comment (nklomp) [inline documents/credential-branding.md]: Again, this part really should not be here. It is way to distracting. This documentation should be about how to setup an issuer. Not about how wallets should handle that information in code

File: documents/vci-front-end.md [hunk 19]
```diff
@@ -0,0 +1,117 @@
...
+    type: HTMLInputTypeAttribute
+    required?: boolean
+    readonlyWhenAbsentInPayload?: boolean
+    defaultValue?: FormFieldValue
+    label?: string
+    labelUrl?: string
+    readonly?: boolean
+    customValidation?: string
+    display?: {
+        checkboxBorderColor?: string
```
Comment (nklomp) [inline documents/vci-front-end.md:88]: Hmzz, this really should have become a checkbox object, but that is out of the scope of this PR

**Review by nklomp — CHANGES_REQUESTED** [review]
See remarks. You need to be more precise in how to name things. Link them to respective specs and/or use the actual definitions from them.

Also still not addressed my comment, about actual steps you need to take, like having a rectangular image without rounded corners, because the software will take care of that, providing the locales.

Also do not go in details how our SDK or wallet renders branding. That has no place in a project where you are trying to setup a demo OID4VC agent and frontend

File: documents/agent-setup.md (hunk 10 above)
Comment (sksadjad) [reply documents/agent-setup.md]: changed this description both here and the next one

File: documents/agent-setup.md (hunk 11 above)
Comment (sksadjad) [reply documents/agent-setup.md]: changed to suggested description

File: documents/agent-setup.md (hunk 12 above)
Comment (sksadjad) [reply documents/agent-setup.md]: changed to the value in oid4vci spec

File: documents/agent-setup.md (hunk 13 above)
Comment (sksadjad) [reply documents/agent-setup.md]: changed to the value in oid4vci spec

File: documents/agent-setup.md (hunk 14 above)
Comment (sksadjad) [reply documents/agent-setup.md]: mentioned both oid4vp and pex spec for this

File: documents/agent-setup.md (hunk 15 above)
Comment (sksadjad) [reply documents/agent-setup.md]: changed to Verifiable Credentials

File: documents/agent-setup.md (hunk 16 above)
Comment (sksadjad) [reply documents/agent-setup.md]: changed to .env.local

File: documents/credential-branding.md (hunk 17 above)
Comment (sksadjad) [reply documents/credential-branding.md]: changed the structure to have ids, also made links in several other places

File: documents/credential-branding.md (hunk 18 above)
Comment (sksadjad) [reply documents/credential-branding.md]: removed this part.

File: documents/overview.md [hunk 20]
```diff
@@ -0,0 +1,42 @@
+<!--suppress HtmlDeprecatedAttribute -->
```
Comment (nklomp) [inline documents/overview.md]: Please move this file to the main README (and remove the current OID4VP demo section with our ugly UI from it). The current demo is mainly about VCI, but also can use VP, so it covers that use case as well.

The reason this needs to move to the main README is to make sure people will actually find this documentation on how to setup everything

File: documents/credential-branding.md [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+      ...
+    }
+  ],
+  ...
+}
+```
+
+When providing image for branding, be sure that:
+1. The image (both `background_image.url` and `logo.url`) are available to every party interacting with the demo.
+2. Providing a rectangular image without rounded corners, the library will take care of that.
```
Comment (nklomp) [inline documents/credential-branding.md]: Please mention that this is about the background image for the credential card. Someone reading this will not automatically know that

File: documents/credential-branding.md [hunk 22]
```diff
@@ -0,0 +1,81 @@
...
+3. You can have locale-specific branding for your credentials. For this to happen you can add a locale to your display array, with a `locale` key representing your desired locale. Read more in [OID4VCI](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#section-10.2.3.1-2.5.1)
```
Comment (nklomp) [inline documents/credential-branding.md]: That still does not really explain it well to a user. You need to explain that you for instance can have locale specific values for the Credential type, or any claim in the credential, so that you can display a credential in the user's locale

File: documents/overview.md [hunk 23]
```diff
@@ -0,0 +1,42 @@
...
+- [SIOP-OID4VP](https://github.com/Sphereon-Opensource/SIOP-OID4VP)
+  - An OpenID authentication library conforming to the Self Issued OpenID Provider v2 (SIOPv2) and OpenID for Verifiable Presentations (OpenID4VP) as specified in the OpenID Connect working group.
+
+And many more open-source libraries.
+You can use this demo for showcasing VCI abilities as well as SIOP-OID4VP abilities.
+
+In this document we're going to show you step by step what you have to do to set your own VCI demo up.
+Note that in order for you to set a VCI demo up and actually receive a credential, you need to have an SSI compatible wallet. Here is a list of wallet's with such capabilities:
+//TODO: complete this list
+- Sphereon
```
Comment (nklomp) [inline documents/overview.md]: Link this to the Sphereon Wallet Github and the Animo wallet Github for now

File: documents/vci-flow.puml [hunk 24]
```diff
@@ -0,0 +1,35 @@
...
+title
+VCI flow
+A scenario for getting a credential in your SSI wallet
+end title
+
+autonumber
+
+participant "User" as user order 0 #ORANGE
+participant "Wallet" as wallet order 1 #RED
+participant "VCI front end" as vci order 2 #BLUE
```
Comment (nklomp) [inline documents/vci-flow.puml]: VCI frontend -> Demo app

File: documents/vci-flow.puml [hunk 25]
```diff
@@ -0,0 +1,35 @@
...
+participant "Agent" as agent order 3 #GREEN
+
+user -> vci: from the list, select a credential that you want to receive
+vci -> agent: fetch credential metadata/requested presentation definition\nand generate a qr code
+vci -> user: show the qr code
+user -> wallet: scan the qr code
+wallet -> wallet: search the credentials inside the wallet\nfor matching with the presentation definition
```
Comment (nklomp) [inline documents/vci-flow.puml]: Whut? That has 0 todo with VCI!

File: documents/vci-flow.puml [hunk 26]
```diff
@@ -0,0 +1,35 @@
...
+wallet -> user: shows user a list of credentials from which the user can select
+user -> wallet: user selects requested credentials
```
Comment (nklomp) [inline documents/vci-flow.puml]: credential type(s)

File: documents/vci-flow.puml [hunk 27]
```diff
@@ -0,0 +1,35 @@
...
+wallet -> agent: wallet sends credentials to the agent
```
Comment (nklomp) [inline documents/vci-flow.puml]: Nope

File: documents/vci-flow.puml [hunk 28]
```diff
@@ -0,0 +1,35 @@
+@startuml
```
Comment (nklomp) [inline documents/vci-flow.puml:1]: Stopped review on this flow diagram, as you are either mixing up things, or you are trying to merge a SIOP flow with a VCI flow, but in the demo's these are really distinct steps, meaning you should not try to do that in one flow diagram.

Given it is called the VCI flow, only stick to VCI steps.

Also mixing VCI steps with SIOP/OID4VP steps, but then only calling everything VCI frontend is very confusing to say the least

**Review by nklomp — CHANGES_REQUESTED** [review]
See additional remarks.

Stopped review on the VCI flow diagram as a lot of lines didn't make sense to me.

You are either mixing up things, or you are trying to merge a SIOP flow with a VCI flow, but in the demo's these are really distinct steps, meaning you should not try to do that in one flow diagram.

Given it is called the VCI flow, only stick to VCI steps.

Also mixing VCI steps with SIOP/OID4VP steps, but then only calling everything VCI frontend is very confusing to say the least

File: documents/vci-flow.puml (hunk 24 above)
Comment (sksadjad) [reply documents/vci-flow.puml]: changed to demo app

File: documents/vci-flow.puml (hunk 25 above)
Comment (sksadjad) [reply documents/vci-flow.puml]: Yeah, I was going for showing the complete thing oid4vp+oid4vci. removed the oid4vp part from this file

File: documents/vci-flow.puml (hunk 26 above)
Comment (sksadjad) [reply documents/vci-flow.puml]: changed

File: documents/vci-flow.puml (hunk 28 above)
Comment (sksadjad) [reply documents/vci-flow.puml:1]: Yeah, I was going for showing the complete thing oid4vp+oid4vci. removed the oid4vp part from this file

File: documents/vci-flow.puml (hunk 27 above)
Comment (sksadjad) [reply documents/vci-flow.puml]: Yeah, I was going for showing the complete thing oid4vp+oid4vci. removed the oid4vp part from this file

File: documents/overview.md (hunk 23 above)
Comment (sksadjad) [reply documents/overview.md]: linked (this is moved to project's readme)

File: documents/credential-branding.md (hunk 22 above)
Comment (sksadjad) [reply documents/credential-branding.md]: mad some changes and explained it in details

File: documents/credential-branding.md (hunk 21 above)
Comment (sksadjad) [reply documents/credential-branding.md]: changed the wording here to have background image

File: documents/overview.md (hunk 20 above)
Comment (sksadjad) [reply documents/overview.md]: moved and merged original readme with this document

File: README.md [hunk 29]
```diff
@@ -1,18 +1,20 @@
 <h2 style="text-align: center; vertical-align: middle">
     <center><a href="https://www.sphereon.com"><img src="https://sphereon.com/content/themes/sphereon/assets/img/logo.svg" alt="Sphereon" width="320" style="vertical-align: middle" ></a></center>
 
-<br>OID4VC Issuer and Verifier Demo      
+<br>OID4VC Issuer
```
Comment (nklomp) [inline README.md]: Why was this changed, because this demo is about VCI, SIOP and OID4VP?

File: README.md [hunk 30]
```diff
@@ -1,18 +1,20 @@
...
 <br>
 <br>
 </h2>
 
 #### This is a demo to showcase our OpenID for Verifiable Credentials libraries and components.
 
 **Please note that this is not intended as production code. It is a relative simple implementation to show how the
-OID4VC technologies, SIOPv2 (peer to peer authentication), OID4VP (Verification), OID4VCI (Issuance) end Presentation
```
Comment (nklomp) [inline README.md:12]: Why was this changed, because this demo is about VCI, SIOP and OID4VP?

File: documents/agent-setup.md [hunk 31]
```diff
@@ -0,0 +1,137 @@
...
+  <a href="https://www.sphereon.com"><img src="https://sphereon.com/content/themes/sphereon/assets/img/logo.svg" alt="Sphereon" width="400"></a>
+  <br>OID4VC demo's agent setup 
+  <br>
+</h1>
+
+---
+
+**Warning: This project still is in very early development. Breaking changes without notice will happen at this point!**
+
+## Table of contents <a name="toc"></a>
```
Comment (nklomp) [inline documents/agent-setup.md]: No need to include links. Markdown can create TOCs from headers

File: documents/agent-setup.md [hunk 32]
```diff
@@ -0,0 +1,137 @@
...
+1. [Agent Configuration Guide](#agent_config_guide)
+    1. [Configuration Directory Structure](#configuration_dir_structure)
```
Comment (nklomp) [inline documents/agent-setup.md]: No need to include `<a>` links. Markdown can use the headers as relative links. Just lowercase them and instead of whitespace use underscores from the top of my head. But your IDE can help with that

File: documents/agent-setup.md [hunk 33]
```diff
@@ -0,0 +1,137 @@
...
+|                    | `sessionManager`      | (optional) An instance of IRPSessionManager from Sphereon's SIOP-OID4VP library managing session state. |
+|                    | `expiresIn`           | (optional) Defines the duration after which the RP's request or session should expire.                  |
+|                    | `eventEmitter`        | (optional) An object used to handle events, enabling asynchronous event-driven programming.             |
+|                    | `didOpts`             | Options related to the Decentralized Identifier (DID) of the verifier.                                  |
+
+#### 5. presentation_definitions <a name="presentation_definitions"></a>
+
+The content of this directory is essential for defining how the agent understands and processes presentation requests. For a more thorough description of presentation_definition in [presentation exchange documentation](https://identity.foundation/presentation-exchange/#presentation-definition), and how it's being used in [OID4VP](https://openid.net/specs/openid-4-verifiable-presentations-1_0.html) spec.
```
Comment (nklomp) [inline documents/agent-setup.md]: Last sentence isn't proper English. You are either missing something like 'refer to' or 'please have a look at the'

File: documents/agent-setup.md [hunk 34]
```diff
@@ -0,0 +1,137 @@
...
+
+#### 6. templates <a name="templates"></a>
+
+This directory includes templates for the dynamic generation of Verifiable Credentials, We're populating the credential using [handlebars](https://handlebarsjs.com/). You can find examples of this usage in this directory. Here is an example of such usage:
```
Comment (nklomp) [inline documents/agent-setup.md]: Please mention that the values here should match values used in forms or that come from a VP

File: documents/agent-setup.md (hunk 33 above)
Comment (nklomp) [inline documents/agent-setup.md]: Explain that the presentation definition is relevant in scenario's where the demo ask a user for it's credentials from the wallet

File: documents/credential-branding.md [hunk 35]
```diff
@@ -0,0 +1,91 @@
...
+<h1 align="center">
+  <br>
+  <a href="https://www.sphereon.com"><img src="https://sphereon.com/content/themes/sphereon/assets/img/logo.svg" alt="Sphereon" width="400"></a>
+  <br>OID4VC demo's credential branding 
+  <br>
+</h1>
+
+---
+
+**Warning: This project still is in very early development. Breaking changes without notice will happen at this point!**
```
Comment (nklomp) [inline documents/credential-branding.md]: Remove these warnings from the individual documents in this dir

File: documents/credential-branding.md [hunk 36]
```diff
@@ -0,0 +1,91 @@
...
+      ...
+    }
+  ],
+  ...
+}
+```
+
+When providing image for branding, be sure that:
+1. The image (both `background_image.url` and `logo.url`) are available to every party interacting with the demo.
+2. Providing a rectangular image for background image of credential cards, without rounded corners, the library will take care of that.
```
Comment (nklomp) [inline documents/credential-branding.md]: Providing -> You will have to provide

File: documents/vci-flow.puml [hunk 37]
```diff
@@ -0,0 +1,26 @@
...
+autonumber
+
+participant "User" as user order 0 #ORANGE
+participant "Wallet" as wallet order 1 #RED
+participant "Demo app" as demo order 2 #BLUE
+participant "Agent" as agent order 3 #GREEN
+
+user -> demo: from the list, selects a credential type that she wants to receive
+demo -> demo: generates a form with the form data in ecosystem config file
+demo -> user: shows user the form to user.
```
Comment (nklomp) [inline documents/vci-flow.puml]: No proper English

File: documents/vci-flow.puml [hunk 38]
```diff
@@ -0,0 +1,26 @@
...
+user -> demo: user fills in the information and clicks on send
+demo -> agent: agent receives the information from demo app and\nwith the credential template, creates a credential offer
+agent -> demo: agent sends the credential offer to the demo app
+demo -> user: demo app shows the credential offer uri in a qr code
+user -> wallet: user scans the qr code with the wallet
+wallet -> user: wallet shows user the credential the she is about to accept
```
Comment (nklomp) [inline documents/vci-flow.puml]: No proper English

File: documents/vci-front-end.md [hunk 39]
```diff
@@ -0,0 +1,133 @@
...
+      "id": "did:jwk:eyJhbGciOiJFUzI1NksiLCJ1c2UiOiJzaWciLCJrdHkiOiJFQyIsImNydiI6InNlY3AyNTZrMSIsIngiOiI3S2hEbmRTZEtGVGlzTy1BNXhwWno1SkVvYkJ2NVdELTAxWmVUbVJ0ZmdvIiwieSI6IjVFR3kyZlZPX1JKZzdsdjUwUDRuRjJ0ZklqakhsNERWOGtDQm1kdHZaQ0UifQ"
+    }
+  },
+...
+}
+```
+
+### Starting the VCI frontend <a name="starting"></a>
+
+Once you've managed to create your own configuration file, you can start you have to alter the value of `REACT_APP_DEFAULT_ECOSYSTEM` in your `.env` file. After this, you can start this module using `start:prod` or `start:dev` scripts.
```
Comment (nklomp) [inline documents/vci-front-end.md]: .env.local

**Review by nklomp — CHANGES_REQUESTED** [review]
See remarks.

- No need to include html in links/TOCs. This is supported by Markdown itself, you just have to follow the rules of replacing spaces with underscores from the TOC. Your IDE can do the autocompletion in Markdown for these relative links to headers
- Nice that the OID4VP is removed from the flow, but obviously OID4VP is being used in parts of the demo, so I would also expect a diagram for that
- Removal of SIOP/OID4VP from description etc makes no sense. My previous comment in the diagram were about trying to mix everything in a single flow diagram, whilst in reality it doesn't work like that in a single interaction

File: documents/vci-front-end.md (hunk 39 above)
Comment (sksadjad) [reply documents/vci-front-end.md]: changed

File: documents/vci-flow.puml (hunk 38 above)
Comment (sksadjad) [reply documents/vci-flow.puml]: changed to `The wallet application reveals to the user the details of the credential they are about to accept.`

File: documents/vci-flow.puml (hunk 37 above)
Comment (sksadjad) [reply documents/vci-flow.puml]: changed to `The demo application displays this form to the user.`

File: documents/credential-branding.md (hunk 36 above)
Comment (sksadjad) [reply documents/credential-branding.md]: changed

File: documents/credential-branding.md (hunk 35 above)
Comment (sksadjad) [reply documents/credential-branding.md]: removed all of them in this dir

File: documents/agent-setup.md (hunk 33 above)
Comment (sksadjad) [reply documents/agent-setup.md]: added the explanation

File: documents/agent-setup.md (hunk 34 above)
Comment (sksadjad) [reply documents/agent-setup.md]: added the explanation

File: documents/agent-setup.md (hunk 33 above)
Comment (sksadjad) [reply documents/agent-setup.md]: changed

File: documents/agent-setup.md (hunk 32 above)
Comment (sksadjad) [reply documents/agent-setup.md]: removed

File: documents/agent-setup.md (hunk 31 above)
Comment (sksadjad) [reply documents/agent-setup.md]: removed

File: README.md (hunk 30 above)
Comment (sksadjad) [reply README.md:12]: added back

File: README.md (hunk 29 above)
Comment (sksadjad) [reply README.md]: added back

**Review by nklomp — APPROVED** [review]
